### PR TITLE
issue 399.1: store only transient link information by IResearchViewCoordinator

### PR DESCRIPTION
**Pull request by vasiliy-arangodb**

### CODE REVIEW

**Comment by vasiliy-arangodb** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/276/

File: arangod/Aql/ExecutionBlock.cpp [hunk 1]
```diff
@@ -54,6 +54,12 @@ static std::string const& stateToString(ExecutionState state) {
   return unknownString;
 }
 
+struct ExecutionBlockTypeHash {
+  size_t operator()(const ::arangodb::aql::ExecutionBlock::Type& value) const {
+    return size_t(value);
```
Comment (graetzer) [inline arangod/Aql/ExecutionBlock.cpp]: is this a suitable hash function?

Comment (dothebart) [reply arangod/Aql/ExecutionBlock.cpp]: its basically a class enum which newer compilers don't understand.

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/

**Pull request merged**